libs/ledger-live-common/src/hw/signMessage/index.ts

```diff
@@ -13,29 +13,28 @@ import { createAction as createAppAction } from "../actions/app";
 import type { Device } from "../actions/types";
 import type { ConnectAppEvent, Input as ConnectAppInput } from "../connectApp";
 import { withDevice } from "../deviceAccess";
-import type { MessageData, Resolver, Result } from "./types";
-import { DerivationMode } from "../../derivation";
+import type { MessageData, SignMessage, Result } from "./types";
+import type { TypedMessageData } from "../../families/ethereum/types";
```

Comment:
Not ideal to have a direct dependency to a family here...
Could `MessageData` be cast as `TypedMessageData` inside the family-specific functions, maybe?
Using `message as TypedMessageData`.

Reply:
`MessageData` and `TypedMessageData` are not exactly equivalent.
I agree we shouldn't import anything specific here, but it would cost a lot more refacto at once and I don't have enough knowledge of the current code base to be sure.
Casting should be use with caution. Someone may later do some breaking change to the definition of the type and the casting code will raise no error.
In our current situation, casting will not help `signMessage::prepareMessageToSign` caller to be aware that the type return may be `TypedMessageData`.